Form/FormMapper.php

```diff
@@ -275,6 +280,21 @@ public function addHelp($name, $help)
     }
 
     /**
+     * Symfony default form class sadly can't handle
+     * form element with dots in its name (when data
+     * get bound, the default dataMapper is a PropertyPathMapper).
+     * So use this trick to avoid any issue.
+     *
+     * @param string $fieldName
+     *
+     * @return string
+     */
+    public function sanitizeFieldName($fieldName)
+    {
+        return str_replace(array('__', '.'), array('____', '__'), $fieldName);
```

Comment:
Sorry, I'm not sure to understand what you mean.
This pr fixes 2 things.
The first one is the mapping between `ChoiceFieldMaskType` map's option, because today the `. ` is replaced by `__` in formMapper and this wasn't reflected there.
The second one is a bug in FormMapper. You can check  it by running this test on master branch in FormMapperTest.
```
public function testFieldNameIsSanitized()
{
    $this->contractor->expects($this->any())
        ->method('getDefaultOptions')
        ->will($this->returnValue(array()));

    $this->formMapper
        ->add('fo.o', 'bar')
        ->add('fo__o', 'foobaz')
    ;

    $this->assertSame(2, count($this->formMapper->keys()));
}
```

Is more clear ?